Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js

```diff
@@ -16,6 +18,292 @@ function checkAgentAuditHistory() {
+    cy.get("div").contains("Required").should("not.exist");
+
+    cy.get("[data-testid=address-lines-input]").clear();
+    cy.get("[data-testid=city-input]").clear();
+    cy.get('[data-testid="country-input"]')
+        .parent()
+        .within(() => {
+            cy.get('button[aria-label="Clear"]').click();
+        });
+    cy.get("body").click(0, 0);
+
+    cy.get("[data-testid=address-confirmation-button]").click();
+
+    validateCovid19InputContainsError("[data-testid=address-lines-input]");
+    validateCovid19InputContainsError("[data-testid=city-input]");
+    validateCovid19InputContainsError("[data-testid=province-input]");
+    validateCovid19InputContainsError("[data-testid=postal-code-input]");
+
+    cy.get("[data-testid=address-cancel-button]").click();
+    cy.get("[data-testid=address-confirmation-form]").should("not.exist");
+}
+
+function validateMailAddressFormSubmission() {
+    cy.get("[data-testid=mail-button]").click();
+    cy.get("[data-testid=address-confirmation-form]").should("be.visible");
+    cy.get("[data-testid=address-lines-input]")
+        .clear()
+        .type("9105 ROTTERDAM PLACE");
+    cy.get("[data-testid=city-input]").clear().type("CLITHEROE");
+    cy.get("[data-testid=country-input]").clear();
+    cy.get(".mud-list .mud-list-item-text").contains("Canada").click();
+    cy.get("[data-testid=province-input]").click();
+    cy.get("[data-testid=province]").contains("British Columbia").click();
+    cy.get("[data-testid=postal-code-input]").clear().type("V3X 4J5");
+    cy.get("[data-testid=address-confirmation-button]").click();
+    cy.get("[data-testid=address-confirmation-form]").should("not.exist");
+}
+
+function validateCovid19TreatmentAssessmentFormBackCancel() {
+    cy.get("[data-testid=start-covid-19-treatment-assessment-button]").click();
+    cy.url().should("include", "/covid-19-treatment-assessment");
+    cy.get("[data-testid=back-icon]").click();
+    cy.url().should("include", "/patient-details");
+    cy.get("[data-testid=start-covid-19-treatment-assessment-button]").click();
+    cy.url().should("include", "/covid-19-treatment-assessment");
+    cy.scrollTo("bottom");
+    cy.get("[data-testid=cancel-covid-19-treatment-assessment]").click();
+    cy.url().should("include", "/patient-details");
+}
+
+function validateCovid19TreatmentAssessmentInfoMessageForRadioSelection() {
+    cy.get("[data-testid=start-covid-19-treatment-assessment-button]").click();
+    cy.get(
+        "[data-testid=assessment-question-1] [data-testid=assessment-option-yes]"
+    ).click();
+    cy.get(
+        "[data-testid=assessment-question-1] [data-testid=treatment-benefit-not-indicated]"
+    ).should("not.exist");
+    cy.get(
+        "[data-testid=assessment-question-1] [data-testid=assessment-option-no]"
+    ).click();
+    cy.get(
+        "[data-testid=assessment-question-1] [data-testid=treatment-benefit-not-indicated]"
+    ).should("be.visible");
```

Comment:
Nit: could be simplified by scoping all commands to the relevant question, e.g.

```
cy.get("[data-testid=assessment-question-1]").within(() => {
    cy.get("[data-testid=assessment-option-yes]").click();
    cy.get("[data-testid=treatment-benefit-not-indicated]").should("not.exist");

    // etc.
});
```

Reply:
Changed